### ISSUE TITLE
[clang] set feature test macro ` __cpp_deduction_guides` for P1814R0

### ISSUE THREAD

**Issue by h-vetinari** [llvm/llvm-project#102847]
The paper (#54051) was mostly implemented in https://github.com/llvm/llvm-project/pull/77890 & #89358, where @cor3ntin [said](https://github.com/llvm/llvm-project/pull/89358#discussion_r1575876411):
> [...] I suspect that we do want to finalize the feature for Clang 19.

I cannot tell from browsing the various PRs/issues what (if anything) is still left to do, so opening this issue to track (AFAICT, there are some follow-up issues (e.g. https://github.com/llvm/llvm-project/issues/92224), but they don't look like they have any user-visible impact).

CC @hokein

**Comment by h-vetinari** [comment]
https://github.com/llvm/llvm-project/issues/55876 might also be closable in the context of this effort (or already now based on the implementation in Clang 19).

**Comment by hokein** [comment]
Yeah, I don't see any remaining big issues for this feature now. We should definitely set this feature flag in clang 20.
But before that, I'd like to get #103016 fixed.

> https://github.com/llvm/llvm-project/issues/55876 might also be closable in the context of this effort (or already now based on the implementation in Clang 19).

This doesn't seem to be related to this feature.

**Comment by cor3ntin** [comment]
@hokein Is this still on your radar?

**Comment by hokein** [comment]
Yeah, it is on my radar. My plan is to wait for #98788 to be landed, and then fix #103016 (which needs some code from #98788).
